### ISSUE TITLE
[STU-167] Undefined array key error in SearchableBehavior.php when replicaIndex is true

### ISSUE THREAD

**Issue by mike-moreau** [studioespresso/craft-scout#303]
I am running Scout 4.1.0 on Craft 4.9.7 and configuring a Replica Index.

My `/config/scout.php` file looks like this:

```
<?php

use craft\helpers\App;
use craft\elements\Entry;
use rias\scout\ScoutIndex;
use rias\scout\IndexSettings;
use craft\elements\db\EntryQuery;
use Modules\Serialization\Transformer;

function createSettings(): IndexSettings
{
    $settings = IndexSettings::create()
        // ...settings methods...
        ->alternativesAsExact(ALTERNATIVES_AS_EXACT);

    return $settings;
}

return [
    'indices' => [
        ScoutIndex::create('Site (English)')
            ->elementType(Entry::class)
            ->transformer([Transformer::class, 'scout'])
            ->criteria(fn (EntryQuery $query) => $query->site('default')->status('live'))
            ->indexSettings(
                createSettings()
                    ->replicas(['virtual(Site (English) (Sorted))'])
            ),
        ScoutIndex::create('Site (English) (Sorted)')
            ->replicaIndex(true)
            // ->criteria(fn (EntryQuery $query) => $query->id(0))
            ->indexSettings(createSettings(true)),
        ScoutIndex::create('Site (Spanish)')
            ->elementType(Entry::class)
            ->transformer([Transformer::class, 'scout'])
            ->criteria(fn (EntryQuery $query) => $query->site('spanish')->status('live'))
            ->indexSettings(
                createSettings()
                    ->replicas(['virtual(Site (Spanish) (Sorted))'])
            ),
        ScoutIndex::create('Site (Spanish) (Sorted)')
            ->replicaIndex(true)
            // ->criteria(fn (EntryQuery $query) => $query->id(0))
            ->indexSettings(createSettings(true)),
    ],
];
```

Based off this [merged PR](https://github.com/studioespresso/craft-scout/pull/275) and [the documentation](https://github.com/studioespresso/craft-scout?tab=readme-ov-file#replicas), I expected the Replica Indexes to not require criteria, like this example:

```
        ScoutIndex::create('Site (English) (Sorted)')
            ->replicaIndex(true)
            ->indexSettings(createSettings(true)),
```

However, I get the following error when saving an entry:

```
PHP Warning – [yii\base\ErrorException](https://www.yiiframework.com/doc-2.0/yii-base-errorexception.html)
Undefined array key 0
In .../vendor/studioespresso/craft-scout/src/behaviors/SearchableBehavior.php

Line 88 && ($criteriaSiteIds[0] === '*' || in_array((int)$this->owner->siteId, $sit

```
<img width="949" alt="Screenshot 2024-06-12 at 7 42 22 AM" src="https://github.com/studioespresso/craft-scout/assets/92817780/819df3c9-5ed5-4bf6-b578-687d70f2a99c">

If I add criteria to the Replica Indexes config, the error goes away. I'm using `$query->id(0)` to ensure an empty Element Query, but would prefer not to have any criteria at all.

```
        ScoutIndex::create('Site (English) (Sorted)')
            ->replicaIndex(true)
            ->criteria(fn (EntryQuery $query) => $query->id(0))
            ->indexSettings(createSettings(true)),
```

Am I misunderstanding the documentation for Replica Indexes or is this a bug perhaps?

<sub>[STU-167](https://linear.app/studio-espresso/issue/STU-167/undefined-array-key-error-in-searchablebehaviorphp-when-replicaindex)</sub>

**Comment by janhenckens** [comment]
Fixed in 4.1.1, will leave this open to make sure I don't forget to the Craft 5 branch

**Comment by janhenckens** [comment]
Out for Craft 5 in 5.0.1.
